### PR TITLE
ID should not be an IRI when not using an hypermedia format, don't hide the raw ID property by default

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #758
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by dunglas** [comment]
This issue raises a more fundamental problem:

With the current implementation, we cannot guess the intent of the user: does he want to hide the `id` property and just generate the `@id` JSON-LD identifier or those he want to include the raw id property (like in the failing test).

I only see one way to fix it without hack:

* Always include the raw `id` property if no normalization group have been specified (the default behavior)
* Use normalization/denormalization groups to hide it explicitly if you want too

Pros:

* Fixes #758
* Swagger will be easier to use with JSON-LD/Hydra (the raw `id` will be there by default)

Cons:

* Add a (maybe unwanted) extra raw `id` property
* Change the behavior comparent to v1 and previous RCs (all entities not using serialization groups will now have an extra `id` property)

WDYT @api-platform/core-team? We need to take a decision before the next RC (planned for this week).

**Comment by dunglas** [comment]
ping @api-platform/core-team

File: src/Serializer/AbstractItemNormalizer.php [hunk 1]
```diff
@@ -407,7 +407,7 @@ protected function getAttributeValue($object, $attribute, $format = null, array
      *
      * @return string|array
      */
-    private function normalizeRelation(PropertyMetadata $propertyMetadata, $relatedObject, string $resourceClass, string $format = null, array $context)
+    protected function normalizeRelation(PropertyMetadata $propertyMetadata, $relatedObject, string $resourceClass, string $format = null, array $context)
```
Comment (theofidry) [inline src/Serializer/AbstractItemNormalizer.php]: Why this change?

**Comment by theofidry** [comment]
👍 otherwise :)

File: src/Serializer/AbstractItemNormalizer.php (hunk 1 above)
Comment (dunglas) [inline src/Serializer/AbstractItemNormalizer.php]: Good catch, I should revert it.
